pkg/common-controller/snapshot_controller.go

```diff
@@ -1095,10 +1136,30 @@ func (ctrl *csiSnapshotCommonController) updateSnapshotStatus(snapshot *crdv1.Vo
 	if updated {
 		snapshotClone := snapshotObj.DeepCopy()
 		snapshotClone.Status = newStatus
+
+		// We need to record metrics before updating the status due to a bug causing cache entries after a failed UpdateStatus call.
```

Comment:
I don't think so. Since this is the first time the controller has seen that the snapshot has been created or is ready, it is okay to record IMO.

The update status will go through on the next reconcile so that the end user sees the snapshot as ready. 

The problem here was that UpdateStatus failed, but the snapshotStore cache recorded readyToUse as true. On the next reconcile loop, this snapshot would hit syncReadySnapshot, and update status with readyToUse=true (without starting a new operation, because readyToUse=true).